### DISCUSSION TITLE
New Soon To Be User - Frigate Hardware Question

### DISCUSSION THREAD

**Discussion by The5eeker** [blakeblackshear/frigate#3458] [Ask A Question]
I'm building a security system using Home Assistant (I haven't started yet, I'm almost done gathering hardware).  I have ordered a "[Dell Otiplex7050 Micro PC Intel i5 @3.20GHz 16GB RAM 256GB SSD - WIN10 & APPS]" to run Home Assistant.  I am planing to install at least 4 maybe up to 6 4K security cameras (like Hikvision ColorVu DS-2CD2387G2-LU) and was initially going to run Blue Iris but now I am considering Frigate because of the lower PC requirements if using a Google Coral (I haven't been able to find one yet).

I'm thinking to order a used Dell Optiplex i7 or similar to run Frigate and then try to connect it to Home Assistant over a gigabit connection.  I'm well versed in Windows but I know nothing about Linux, I have reviewed a "how to" on installing Debian 11 and Docker and the install seems daunting.  I'm not sure if I want to try to tackle it or not.  I have read that Home Assistant can run Frigate, it can be installed somewhat easily in Home Assistant, I think. . . .

So.  My questions are as follows:

1. Is a used Dell Optiplex 9020 mini tower or similar good hardware to use for Debian 11, I don't want to spend more than $200-$300.  I don't know if Debian 11 has drivers for that hardware.  If not, what decent system would be ready to use with Debian 11 that can handle 6 4K cameras without costing a fortune?

2. In your experience would the "Dell Otiplex 7050 Micro PC Intel i5 @3.20GHz 16GB RAM 256GB SSD" be powerful enough to run Home Assistant and Frigate with up to 6 4K cameras without maxing it out?  If I go the route of installing Frigate in Home Assistant.

Thanks for your help.

**Comment by NickM-27** [comment]
1. I think it is important to separate out Frigate's hardware requirements so it is easier to understand what systems can and can't perform well. Frigate's main processing demands fall into three categories: video decode, motion detection, and object detection. The coral makes object detection no sweat for the CPU. video decode can be handled on CPU but does require a lot of horsepower, meanwhile a CPU with an iGPU or a dedicated GPU make this work better. I believe the intel CPUs in the machines you mentioned are supported for video decode in ffmpeg but may want to look further into that. The final concern being motion detection which must be run on the CPU, this heavily depends on the resolution, if the docs are followed then and 720p or max 1080p are used for detection I think the devices you linked should be able to handle that fine.
2. Home assistant doesn't really take much processing power on its own, so I think this should be fine. The main concern with installing Frigate as an HA addon is that it is difficult to have it use outside storage other than the main drive. 256GB is not going to hold very much 24/7 footage, so this would be a conern

**Reply by The5eeker** [reply]
Thank you for the answers.  That helps a lot.  

I may go the route of installing Frigate in Home Assistant.  I'm still waiting on the micro PC to be delivered.  I don't know if the SSD in it is a 3.5" SSD or an M.2 NVME, I'm guessing 3.5" probably. . .   I'm thinking to install Home assistant on a 1TB M.2 NVME or something.  I have read that a 8TB WD Purple 3.5 HD can be used in the Optiplex 7050 for video storage, I was going to go that route, but you say it is difficult to store video on a drive outside of the Home Assistant drive?  I can't just point it to the 8TB drive and have it record there?

**Reply by NickM-27** [reply]
If you're running Home assistant OS with frigate inside on the machine then it's not easy to do that, running HASS in docker (or HAOS in VM) and frigate in docker it is very easy

**Reply by The5eeker** [reply]
Hummm.  It seems there is difficulty either way, easy install, hard storage; difficult install easy storage. . .   I may get a 2TB NVMe and just record to that or something.  I'll have to think about it a while longer, it will be another month or so before I will be buying a camera, I'll buy some little by little.

Does anyone know some links to good hand holding, step by step for installing Debian 11 with docker and getting Frigate running and using it with Home assistant?  In case I do decide to go that route.  I know nothing about Linux or how to use it, so whatever I do will have to be based on step by step tutorials. (type this, do that, this is what it means, this is what it does, etc.).  

What used PC will run Debian, docker, Frigate without any fuss, like needing certain hardware or something because there are no Linux drivers for this or that type of PC. . . .

Thanks for the help NickM-27.
